### ISSUE TITLE
beta branch not on flathub-beta

### ISSUE THREAD

**Issue by mikelpr** [flathub/org.blender.Blender#23]

**Comment by bochecha** [comment]
Yes, that's because ARM and Aarch64 don't build and I haven't had time to investigate and fix it yet.

https://flathub.org/builds/#/builders/51/builds/3/steps/6/logs/stdio
Any help would be appreciated, especially since my computer just broke and I won't be able to work on this until I fix it or buy a new one. 🙁

**Comment by mikelpr** [comment]
is that a blender bug? does blender stable build on them?

**Comment by bochecha** [comment]
> is that a blender bug?

As I said in my previous comment, I don't know and haven't had time to investigate. And as I mentioned, it is unlikely I will be able to any time soon.

I gave the link to the build logs so others, like you, could investigate. If you care about this issue, please help.

> does blender stable build on them?

Yes.

**Comment by bochecha** [comment]
Some preliminary investigation: it is indeed Blender which fails to build on Aarch64 and ARM, not one of its dependencies. (they all built successfully)

This is the failure;

```
[200/3646] Building CXX object intern/cycles/device/CMakeFiles/cycles_device.dir/device_cpu.cpp.o
FAILED: intern/cycles/device/CMakeFiles/cycles_device.dir/device_cpu.cpp.o 
/usr/bin/c++  -DBOOST_ALL_NO_LIB -DCCL_NAMESPACE_BEGIN="namespace ccl {" -DCCL_NAMESPACE_END=} -DCYCLES_GFLAGS_NAMESPACE=gflags -DGLEW_NO_ES -DGLEW_NO_GLU -DGLEW_STATIC -DGOOGLE_GLOG_DLL_DECL="" -DNDEBUG -DWITH_BLENDER_GUARDEDALLOC -DWITH_CUDA -DWITH_CUDA_DYNLOAD -DWITH_CYCLES_LOGGING -DWITH_GL_PROFILE_CORE -DWITH_MULTI -DWITH_OCIO -DWITH_OPENCL -DWITH_OPENGL -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D__LITTLE_ENDIAN__ -I../intern/cycles/../atomic -I../intern/cycles/device/.. -I../intern/cycles/device/../../glew-mx -I../intern/cycles/device/../../../extern/cuew/include -isystem ../extern/glog/include -isystem ../extern/gflags/src -isystem /app/include -isystem /app/include/OpenImageIO -isystem /app/include/OpenEXR -isystem ../extern/glew/include -isystem ../intern/cycles/device/../../../extern/clew/include -Wredundant-decls -Wall -Wno-invalid-offsetof -Wno-sign-compare -Wlogical-op -Winit-self -Wmissing-include-dirs -Wno-div-by-zero -Wtype-limits -Werror=return-type -Werror=implicit-function-declaration -Wno-char-subscripts -Wno-unknown-pragmas -Wpointer-arith -Wunused-parameter -Wwrite-strings -Wundef -Wformat-signedness -Wrestrict -Wuninitialized -Wundef -Wmissing-declarations -Wimplicit-fallthrough=5 -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fuse-ld=gold -fopenmp -std=c++11 -pipe -fPIC -funsigned-char -fno-strict-aliasing -fno-trapping-math -fno-math-errno -fno-signed-zeros -fno-signaling-nans -fno-rounding-math -Werror=float-conversion -Werror=double-promotion -Wno-error=unused-macros -O2 -g -DNDEBUG -MD -MT intern/cycles/device/CMakeFiles/cycles_device.dir/device_cpu.cpp.o -MF intern/cycles/device/CMakeFiles/cycles_device.dir/device_cpu.cpp.o.d -o intern/cycles/device/CMakeFiles/cycles_device.dir/device_cpu.cpp.o -c ../intern/cycles/device/device_cpu.cpp
../intern/cycles/device/device_cpu.cpp: In member function ‘void ccl::CPUDevice::path_trace(ccl::DeviceTask&, ccl::RenderTile&, ccl::KernelGlobals*)’:
../intern/cycles/device/device_cpu.cpp:764:27: error: ‘_MM_FLUSH_ZERO_ON’ was not declared in this scope
   _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
                           ^~~~~~~~~~~~~~~~~
../intern/cycles/device/device_cpu.cpp:764:3: error: ‘_MM_SET_FLUSH_ZERO_MODE’ was not declared in this scope
   _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
   ^~~~~~~~~~~~~~~~~~~~~~~
../intern/cycles/device/device_cpu.cpp:765:31: error: ‘_MM_DENORMALS_ZERO_ON’ was not declared in this scope
   _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
                               ^~~~~~~~~~~~~~~~~~~~~
../intern/cycles/device/device_cpu.cpp:765:3: error: ‘_MM_SET_DENORMALS_ZERO_MODE’ was not declared in this scope
   _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

These seem to be related to SSE, so maybe ARM/Aarch64 don't have that?

**Comment by bochecha** [comment]
Paging @brechtvl who helped a lot in the past. :slightly_smiling_face: 

Hi! Did you see anything like that upstream? Any idea how to fix it?

**Comment by brechtvl** [comment]
Fix committed upstream now:
https://developer.blender.org/rBf08191a4596

**Comment by bochecha** [comment]
Oh wow, that's not at all what I expected! (I thought it would be me building something wrong)

Thanks a lot!

**Comment by bochecha** [comment]
This issue should now be fixed with the latest push to #22.

Build is in progress: https://flathub.org/builds/#/builders/32/builds/759

**Comment by mikelpr** [comment]
it's still not on flathub-beta
